beacon-chain/rpc/beaconv1/pool_test.go

```diff
@@ -0,0 +1,176 @@
+package beaconv1
+
+import (
+	"context"
+	"testing"
+
+	ptypes "github.com/gogo/protobuf/types"
+	"github.com/golang/protobuf/proto"
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1"
+	ethpb_alpha "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	mock "github.com/prysmaticlabs/prysm/beacon-chain/blockchain/testing"
+	"github.com/prysmaticlabs/prysm/beacon-chain/operations/slashings"
+	"github.com/prysmaticlabs/prysm/beacon-chain/operations/voluntaryexits"
+	"github.com/prysmaticlabs/prysm/shared/testutil"
+	"github.com/prysmaticlabs/prysm/shared/testutil/assert"
+	"github.com/prysmaticlabs/prysm/shared/testutil/require"
+)
+
+func TestServer_ListPoolAttesterSlashings(t *testing.T) {
+	ctx := context.Background()
+	beaconState, privKeys := testutil.DeterministicGenesisState(t, 64)
+	slashingsInPool := make([]*ethpb_alpha.AttesterSlashing, 20)
+	v1Slashings := make([]*ethpb.AttesterSlashing, 20)
+	for i := 0; i < len(slashingsInPool); i++ {
+		sl, err := testutil.GenerateAttesterSlashingForValidator(beaconState, privKeys[i], uint64(i))
+		require.NoError(t, err)
+		slashingsInPool[i] = sl
+		marshaledSlashing, err := sl.Marshal()
+		require.NoError(t, err)
+		v1Slashing := &ethpb.AttesterSlashing{}
+		require.NoError(t, proto.Unmarshal(marshaledSlashing, v1Slashing))
```

Comment:
This conversion can benefit from reusing the code if you put it in the migration package.